staging/src/k8s.io/dynamic-resource-allocation/controller/controller.go

```diff
@@ -752,12 +752,20 @@ func (ctrl *controller) syncPodSchedulingContexts(ctx context.Context, schedulin
 	// and shouldn't, because those allocations might have to be undone to
 	// pick a better node. If we don't need to allocate now, then we'll
 	// simply report back the gather information.
+	//
+	// We shouldn't assume that the scheduler has included the selected node
+	// in the list of potential nodes. Usually it does, but let's make sure
+	// that we check it.
+	selectedNode := schedulingCtx.Spec.SelectedNode
+	potentialNodes := schedulingCtx.Spec.PotentialNodes
+	if selectedNode != "" && !hasString(potentialNodes, selectedNode) {
+		potentialNodes = append(potentialNodes, selectedNode)
```

Comment:
Question: Some of the logic below seems to rely on the fact that `selectedNode` is the first element in a slice. Does this mean that we need to prepend it here instead of appending it?

Reply:
No. After `UnsuitableNodes`, the selected node is either the first element, the last element, not present, or in the middle. By ensuring that the truncated element (first or last) is *not* the selected node, we can be sure that the rest includes it regardless where exactly it is.